xarray/core/groupby.py

```diff
@@ -259,11 +259,8 @@ def __init__(self, obj, group, squeeze=False, grouper=None, bins=None,
                 # TODO: sort instead of raising an error
                 raise ValueError('index must be monotonic for resampling')
             s = pd.Series(np.arange(index.size), index)
```

Comment:
Could you make this object `s` inside the helper function instead? It's not needed outside here